documentcloud.php

```diff
@@ -25,73 +25,104 @@
+	 */
     function check_dc_plugin_conflict() {
-        if (is_plugin_active('navis-documentcloud/navis-documentcloud.php')) {
-            add_action( 'admin_notices', array(&$this, 'dc_conflict_admin_notice'));
+        if ( is_plugin_active( 'navis-documentcloud/navis-documentcloud.php' ) ) {
+            add_action( 'admin_notices', array( $this, 'dc_conflict_admin_notice' ) );
         }
     }
 
+	/**
+	 * Create an admin notice when conflicts exist with Navis Documentcloud.
+	 */
     function dc_conflict_admin_notice() {
         ?>
         <div class="error">
-            <p><?php _e( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud-plugin-conflict' ); ?></p>
+            <p><?php echo wp_kses( __( '<b>Warning!</b> You have two conflicting DocumentCloud plugins activated. Please deactivate Navis DocumentCloud, which has been replaced by <a target="_blank" href="https://wordpress.org/plugins/documentcloud/">DocumentCloud</a>.', 'documentcloud' ) ); ?></p>
         </div>
         <?php
     }
 
-    // Register ourselves as an oEmbed provider. WordPress does NOT cURL the 
-    // resource to inspect it for an oEmbed link tag; we have to tell it what 
-    // our oEmbed endpoint looks like.
+	/**
+	 * Register ourselves as an oEmbed provider. WordPress does NOT cURL the
+	 * resource to inspect it for an oEmbed link tag; we have to tell it what
+	 * our oEmbed endpoint looks like.
+	 */
     function register_dc_oembed_provider() {
-    /*
-        Hello developer. If you wish to test this plugin against your
-        local installation of DocumentCloud (with its own testing
-        domain), set the OEMBED_PROVIDER and OEMBED_RESOURCE_DOMAIN
-        constants above to your local testing domain. You'll also want
-        to uncomment the next line to let WordPress connect to local
-        domains.
-    */
-        // add_filter( 'http_request_host_is_external', '__return_true');
-
-        wp_oembed_add_provider("http://"  . WP_DocumentCloud::OEMBED_RESOURCE_DOMAIN . "/documents/*",  WP_DocumentCloud::OEMBED_PROVIDER);
-        wp_oembed_add_provider("https://" . WP_DocumentCloud::OEMBED_RESOURCE_DOMAIN . "/documents/*",  WP_DocumentCloud::OEMBED_PROVIDER);
+		/*
+			Hello developer. If you wish to test this plugin against your
+			local installation of DocumentCloud (with its own testing
+			domain), set the OEMBED_PROVIDER and OEMBED_RESOURCE_DOMAIN
+			constants above to your local testing domain. You'll also want
+			to add the following line to your theme to let WordPress connect to local
+			domains:
+
+			add_filter( 'http_request_host_is_external', '__return_true');
+		*/
+
+        $oembed_resource_domain = apply_filters( 'documentcloud_oembed_resource_domain', WP_DocumentCloud::OEMBED_RESOURCE_DOMAIN );
```

Comment:
This is intended so people can update this within their theme if needed for testing rather than comment it in the plugin code. It's more of a WordPress way to do. Also if they were testing on a VIP preprod site for instance, editing this code would not be an option.